solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -0,0 +1,130 @@
+"""
+Provides preprocessing steps to be performed on the timeseries data.
+"""
+
+
+def preprocess(preprocess_func):
+    """Preprocessing decorator."""
+    # TODO implement this decorator and use it for the rest of
+    # the methods here.
+    raise NotImplementedError
+
+
+def exclude(values, quality_flags=None, **kwargs):
+    """
+    Return a timeseries with all questionable values removed.
+
+    All NaN values will be removed first then iff quality_flag is set the
+    corresponding values will also be removed.
+
+    Parameters
+    ----------
+    values : pd.Series
+        Timeseries values
+    quality_flags : pd.Series
+        Timeseries quality flags, Default None
+
+    Returns
+    -------
+    pd.Series :
+        Timeseries of values excluding questionable values.
+
+    Raises
+    ------
+    SfaMetricsInputEror if the `values` and `quality_flags` do not match.
+    """
+
+    # Missing values
+    bad_idx = values.isna()
+
+    # Handle quality flags
+    if quality_flags is not None:
+        bad_quality_idx = (quality_flags == 1)
+        bad_idx = bad_idx | bad_quality_idx
+
+    # print(bad_idx)
+    return values[~bad_idx]
+
+
+def fill_static_value(values, quality_flags, default_value, threshold):
```

Comment:
let's wait on the functions here and below. ok with me to leave them raising `NotImplementedError`.

Reply:
Sounds good. I guess we'll have a better idea idea after the stakeholder meeting next week.